src/net_processing.cpp

```diff
@@ -864,6 +906,10 @@ void PeerLogicValidation::ReattemptInitialBroadcast(CScheduler& scheduler) const
 }
 
 void PeerLogicValidation::FinalizeNode(NodeId nodeid, bool& fUpdateConnectionTime) {
+    {
+        LOCK(g_peer_mutex);
+        g_peer_map.erase(nodeid);
```

Comment:
Should this and ```mapNodeState.erase(nodeid)``` happen at the same time, under the same cs_main lock?

What are the consequences of them being out of sync? (Still reachable by ```State()``` but not by ```GetPeerRef()```)

Reply:
No bad consequences, but I've moved the Peer cleanup underneath the cs_main lock anyway to make this simpler to think about.